contracts/SynthSummaryUtil.sol

```diff
@@ -1,37 +1,91 @@
     function balanceOf(address owner) external view returns (uint);
     function totalSupply() external view returns (uint);
 }
-contract ISynthetix {
-    ISynth[] public availableSynths;
-    function availableSynthCount() public view returns (uint);
-    function availableCurrencyKeys()
-        public
-        view
-        returns (bytes32[]);
+interface ISynthetix {
+    function availableSynths(uint index) external view returns (ISynth);
+    function availableSynthCount() external view returns (uint);
+    function availableCurrencyKeys() external view returns (bytes32[] memory);
 }
 
-contract IExchangeRates {
+interface IExchangeRates {
     function rateIsFrozen(bytes32 currencyKey) external view returns (bool);
-    function ratesForCurrencies(bytes32[] currencyKeys) external view returns (uint[] memory);
+    function ratesForCurrencies(bytes32[] calldata currencyKeys) external view returns (uint[] memory);
     function effectiveValue(bytes32 sourceCurrencyKey, uint sourceAmount, bytes32 destinationCurrencyKey)
-    public
+    external
     view
     returns (uint);
 }
 
-contract SynthSummaryUtil {
-    ISynthetix public synthetix;
-    IExchangeRates public exchangeRates;
-    constructor(address _synthetix, address _exchangeRates) public {
-        synthetix = ISynthetix(_synthetix);
-        exchangeRates = IExchangeRates(_exchangeRates);
+interface IAddressResolver {
+    function getAddress(bytes32 name) external view returns (address);
+    function getSynth(bytes32 key) external view returns (address);
+    function requireAndGetAddress(bytes32 name, string calldata reason) external view returns (address);
+}
+
+
+contract Owned {
+    address public owner;
+    address public nominatedOwner;
+
+    constructor(address _owner) public {
+        require(_owner != address(0), "Owner address cannot be 0");
+        owner = _owner;
+        emit OwnerChanged(address(0), _owner);
+    }
+
+    function nominateNewOwner(address _owner) external onlyOwner {
+        nominatedOwner = _owner;
+        emit OwnerNominated(_owner);
+    }
+
+    function acceptOwnership() external {
+        require(msg.sender == nominatedOwner, "You must be nominated before you can accept ownership");
+        emit OwnerChanged(owner, nominatedOwner);
+        owner = nominatedOwner;
+        nominatedOwner = address(0);
+    }
+
+    modifier onlyOwner {
+        require(msg.sender == owner, "Only the contract owner may perform this action");
+        _;
+    }
+
+    event OwnerNominated(address newOwner);
+    event OwnerChanged(address oldOwner, address newOwner);
+}
+
+
+contract SynthSummaryUtil is Owned {
+
+    IAddressResolver public addressResolver;
+
+    bytes32 internal constant CONTRACT_SYNTHETIX = "Synthetix";
+    bytes32 internal constant CONTRACT_EXRATES = "ExchangeRates";
+    bytes32 internal constant SUSD = "sUSD";
+
+    constructor(address resolver) public Owned(msg.sender) {
```

Comment:
if you initiate with https://contracts.synthetix.io/ReadProxyAddressResolver, then you don't need owner

Reply:
Good point. I actually did this so not sure why I left this in.